internal/bffauth/client.go

```diff
@@ -0,0 +1,202 @@
+package bffauth
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/bluesky-social/indigo/atproto/crypto"
+)
+
+type ExternalHabitatUser struct {
+	DID       string
+	PublicKey crypto.PublicKey
+	Host      string
+}
+
+func getTempStores() (map[string]crypto.PrivateKey, map[string]*ExternalHabitatUser, error) {
+	testKey, err := crypto.GeneratePrivateKeyP256()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	testPubKey, err := testKey.PublicKey()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	testUser := &ExternalHabitatUser{
+		DID:       "did:plc:avkzrsj7u4kmoq33gx6v4lh4",
+		PublicKey: testPubKey,
+		Host:      "beacon-dev.tail07d32.ts.net",
+	}
+
+	return map[string]crypto.PrivateKey{
+			"did:plc:avkzrsj7u4kmoq33gx6v4lh4": testKey,
+		}, map[string]*ExternalHabitatUser{
+			"did:plc:avkzrsj7u4kmoq33gx6v4lh4": testUser,
+		}, nil
+
+}
+
+type Client struct {
+	// This is the store of temporary friends that are added to the BFF.
+	// Helps us avoid implementing full public key resolution from DIDs with TailScale / localhost  setups
+	tempFriendStore map[string]*ExternalHabitatUser
+
+	clientDID string
+
+	privateKey crypto.PrivateKey
+
+	activeTokens map[string]string
+}
+
+func NewClient(clientDID string) (*Client, error) {
```

Comment:
Let's replace `tempFriendStore` with a HabitatResolver and a PublicKeyResolver 
So somewhere in internal maybe in `internal/types?` we have

```
type HabitatResolver func(did string) (habitatHost string, err error)
type PublicKeyResolver func(did string) (crypto.PublicKey, err error)
```

Then your `NewClient` can take these in and you can create some mock implementations for testing only. That will get rid of `tempFriendStore`